### ISSUE TITLE
Global default for scope

### ISSUE THREAD

**Issue by Sytten** [caido/caido#367]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help us know what to work on
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Description

Right now each tool can use a different scope with none selected by default, we should offer a way to have a default scope for all tools.

### Use cases

Saves some clicks to the user.

**Comment by Sytten** [comment]
Love the comment @andreystepanov !!
Global scope definition is interesting, so is the fact that a lot of put of scope elements we could even not store in the database.

I am wondering if we might have two different concepts that would be worth splitting. One seems more like domains for a specific target, usually that correlates with a specific project. The other is global exclusions for stuff you just dont want to see nor keep.

**Comment by andreystepanov** [comment]
> I am wondering if we might have two different concepts that would be worth splitting. One seems more like domains for a specific target, usually that correlates with a specific project. The other is global exclusions for stuff you just dont want to see nor keep.

I don't think that we have two different concepts here.
They are all related to `Scope` of the project hacker is working on.

The only difference is that `Global Scope` is carried across all of the projects that has been created by the hacker.
And i would like to point out that it applies not to only `Out Of Scope` items, but also for `In Scope` items.

As in my example above, i would like to keep track of requests to all popular cloud/SaaS services out there, without being configured on per-project basis, rather being configured only once. 

I just want to clarify that we are on the same page here about functionality :)

As a simple improvements to current UI screen, you can consider adding some sort of checkbox to the `Scope presets`, that will indicate that all listed `In Scope` and/or `Out Of Scope` ips/hosts are saved as `Global Scope`.

I'm not very familiar with data structures/schema you guys use to store scope items in the database, but potentially `Global Scope` might be stored separately from the project itself, and then incoming requests can be checked against both `Global Scope` and `Project Scope` (both objects/arrays can be merged at runtime with `Project Scope` taking precedence when duplicate encountered).

Also worth pointing out that there must be an option to override `Global Scope` for specific projects.
So that any ips/hosts inside `Project Scope` configuration has higher precedence that `Global Scope`.

Another tricky part might be syncing modified `Global Scope` with existing projects, especially if you decide to store a copy of `Global Scope` at the time project gets created. This is why, probably, storing `Global Scope` separately from projects would be beneficial and more flexible in the long run.

But you would still need to account for corner cases, such as when you:

1. Create `Global Scope` with `%.amazonaws.com`.
2. Create project with `Project Scope`.
3. Accumulate lot's of requests by fiddling with an application/website. 
4. Days/weeks/months go by and you decided to modify `Global Scope` by removing `%.amazonaws.com`.
5. Now you need to delete all stored requests to/from `%.amazonaws.com` across all projects to reflect current changes.
   You can probably do it when project launches, or better proactively, right after `Global Scope` gets modified. It all depends on an actual architecture/schema.
   
Hope that helps.
Feel free to ask for clarifications ;)

**Comment by Sytten** [comment]
Thanks a lot for the detailed explanation @andreystepanov, what you explained is definitely for the issue https://github.com/caido/caido/issues/493 I linked above. Both are useful and I agree that the global implementation could be tricky, but lots of good idea.
